lib/user.rb

```diff
@@ -0,0 +1,28 @@
+require './lib/joke'
+require 'pry'
+class User
+attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(insert_joke)
+    @jokes << (insert_joke)
+  end
+
+  def tell(listener, joke)
+    listener.jokes << joke
+  end
+
+  def joke_by_id(desired_id)
+  
+    jokes.find do |ids|
+      ids = desired_id
+      @jokes == (desired_id)
+      return @jokes.at(ids - 1)
+    end
```

Comment:
1. calling your block parameter `ids` implies that each thing in your `jokes` array is another array of `id` values
2. `@jokes == desired_id` isn't going to work if your desired_id is an integer and `@jokes` is an array of Joke objects
3. @jokes.at(index_position) would be fine but we should never guarantee that we're *always* injecting Joke objects into our `@jokes` array in a perfectly sequential numeric order

Using `.find` is the right call here, but it just needs a boolean expression inside to know if it hit a match. In this case a setup like this would suffice:
```ruby
@jokes.find do |joke|
  joke.id == desired_id
end
```